cmd/aws-env/main.go

```diff
@@ -214,7 +216,43 @@ func invoke(r *awsenv.Replacer, prog string, args []string) error {
 	cmd.Stdin = os.Stdin
 	cmd.Stdout = os.Stdout
 	cmd.Stderr = os.Stderr
-	return cmd.Run()
+
+	// in order to make sure that we catch and propagate signals correctly, we need
+	// to decouple starting the command and waiting for it to complete, so we can
+	// send signals as it runs
+	err = cmd.Start()
+	if err != nil {
+		log.WithError(err).Error("failed to start child process")
+		return err
+	}
+
+	// wait for the command to finish
+	errCh := make(chan error, 1)
+	go func() {
+		errCh <- cmd.Wait()
+		close(errCh)
+	}()
+	sigCh := make(chan os.Signal, 1)
+	signal.Notify(sigCh)
+
+	for {
```

Comment:
This loop isn't necessary, every case ends with a `return`.

Reply:
nevermind, this so that we pass along all signals even non terminating ones?

Reply:
Correct